### PR TITLE
fix: will fail when a coinbase tx is not confirmed

### PR DESCRIPTION
**Pull request by FriendsFerdinand**
Issue: `as-segwit-tx-mined-compact` should fail if a coinbase transaction cannot be confirmed to be included in a block.

Fix:  verifying a transaction fails when a transaction verification returns an `(ok false)`.

### CODE REVIEW

**Review by MarvinJanssen — APPROVED** [review]
Let's get this merged.

cc @friedger

**Comment by jo-tm** [comment]
We need 2 reviews from @MarvinJanssen @friedger @setzeus

**Comment by jo-tm** [comment]
work in progress.

**Comment by MarvinJanssen** [comment]
@FriendsFerdinand can we bring the generation script to a separate PR? Then we can merge the clarity-bitcoin fix already.

File: sbtc-mini/tests/clarity-bitcoin_test.clar [hunk 1]
```diff
@@ -230,7 +194,8 @@
       (list 0x3d52480061d7634fa8060430cf86d8de3f577499a2056f5ff80cc36918a78dcc 0x41dd33b4cffe074cc8263f98da7c81006521eb2cc8712028fa491efed619cffb)
     )
       ok-res (err u1)
-      err-res (if (is-eq err-res ERR-VARSLICE-TOO-LONG) (ok err-res) (err err-res))
+      ;; err-res (if (is-eq err-res ERR-VARSLICE-TOO-LONG) (ok err-res) (err err-res))
+      err-res (begin (print { err-res: err-res }) (ok err-res))
```
Comment (friedger) [inline sbtc-mini/tests/clarity-bitcoin_test.clar:198]: Do we expect a particular error here?

**Review by friedger — APPROVED** [review]
LGTM

The change to bitcoin library looks good.

We can keep the proof generator code as work in progress in the code base.

**Comment by friedger** [comment]
Unit tests needs to be fixed @FriendsFerdinand

**Review by friedger — CHANGES_REQUESTED** [review]
Unit tests fail

**Comment by MarvinJanssen** [comment]
The tests that are failing are coming from the peg-in processor. I think that merging main will resolve the failing tests.

File: sbtc-mini/tests/clarity-bitcoin_test.clar (hunk 1 above)
Comment (FriendsFerdinand) [reply sbtc-mini/tests/clarity-bitcoin_test.clar:197]: Yes, the output is too long to be parsed. Uncommented.

**Comment by MarvinJanssen** [comment]
@friedger waiting on your approval.

**Review by friedger — APPROVED** [review]
LGTM

**Pull request merged**